config/config.go

```diff
@@ -1978,6 +1981,22 @@ func buildSerializer(name string, tbl *ast.Table) (serializers.Serializer, error
 		}
 	}
 
+	if node, ok := tbl.Fields["influx_timestamp_units"]; ok {
+		if kv, ok := node.(*ast.KeyValue); ok {
+			if str, ok := kv.Value.(*ast.String); ok {
+				timestampVal, err := time.ParseDuration(str.Value)
+				if err != nil {
+					return nil, fmt.Errorf("Unable to parse influx_timestamp_units as a duration, %s", err)
+				}
+				// now that we have a duration, truncate it to the nearest
+				// power of ten (just in case)
+				nearestExponent := int64(math.Log10(float64(timestampVal.Nanoseconds())))
```

Comment:
It feels like this should be easier to do, but I think you have the simplest approach here

Reply:
This is direct copy/paste from the other PR. So I'm just going to leave it like this.